src/Identity/Controllers/SsoController.cs

```diff
@@ -60,12 +64,17 @@ public async Task<IActionResult> PreValidate(string domainHint)
                 var requestPath = $"/Account/PreValidate?domainHint={domainHint}&culture={culture}";
                 var httpClient = _clientFactory.CreateClient("InternalSso");
                 using var responseMessage = await httpClient.GetAsync(requestPath);
+
                 if (responseMessage.IsSuccessStatusCode)
                 {
-                    // All is good!
-                    return new EmptyResult();
+                    var res = responseMessage.Content.ReadAsStringAsync().Result;
```

Comment:
Please don't use the `.Result` pattern with `async` as it can cause deadlocks or other threading issues.
```suggestion
                    var res = await responseMessage.Content.ReadAsStringAsync();
```